tests/module_test.go

```diff
@@ -0,0 +1,120 @@
+package testing
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/hashicorp/vault/api"
+)
+
+func TestModule(t *testing.T) {
+	files, err := ioutil.ReadDir("./")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, f := range files {
+		// look for directories with test cases in it
+		if f.IsDir() {
+			if f.Name() != "vendor" {
+				testFiles, testErr := ioutil.ReadDir(f.Name())
+				if testErr != nil {
+					log.Fatal(testErr)
+				}
+
+				// see if a prereq directory exists
+				for _, testF := range testFiles {
+					if testF.IsDir() {
+						if testF.Name() == "prereq" {
+							directory := f.Name() + "/" + testF.Name()
+							runTerraformPreReq(t, directory)
+						}
+					}
+				}
+
+				// run terraform code
+				runTerraform(t, f.Name())
+			}
+		}
+	}
+}
+
+// The prequisite function runs the terraform code but doesn't destroy it afterwards so that the state can be used for further testing
```

Comment:
If the resources are not destroyed when running the test, how do they get destroyed so that cruft is not left laying around? Or, is this just a real old version of the test framework? If old, get the newest from the cookiecutter... https://github.com/plus3it/cookiecutter-tfmodule/tree/master/%7B%7Bcookiecutter.module_name%7D%7D/tests

Reply:
this is the old version of the test framework. Will update to the new one.